orca-queue-tck/src/main/kotlin/com/netflix/spinnaker/orca/q/QueueIntegrationTest.kt

```diff
@@ -679,6 +682,40 @@ class QueueIntegrationTest {
       assertThat(stageByRef("2b").status).isEqualTo(SUCCEEDED)
     }
   }
+
+  @Test
+  fun `stages with synthetic failure stages will run those before terminating`() {
```

Comment:
Added a new integration test to ensure the on failure stages run as expected and everything is in the right state at the end.

Reply:
Solid, thanks!